miageWebApp/src/app/services/cours.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { rawClasses } from '../../data/classes-list';
+import { Cours } from '../model/cours';
+import { Unit } from '../model/unit';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CoursService {
+
+  constructor() { }
+
+  getCours(): Cours[] {
+    // retournes l'ensemble des cours
+    // const listClasses = [];
+    // rawClasses.forEach(c =>
+    //   const classe: Cours = new Cours(c.id, c.label, c.period, c.nbHour, c.teacher, c.detail);
+    //   listClasses.push(classe);
+    // });
+    // return listClasses;
+    rawClasses.map(c => new Cours(c.id, c.label, c.period, c.nbHour, c.teacher, c.detail));
+    return rawClasses;
+  }
+
+  deleteAllClasses(): Cours[] {
```

Comment:
for the moment do not do the delete functions in the service.
remove the element from your array in the component, just to refresh the display

Reply:
corrected